packages/services/src/balances/sip10-balances.service.ts

```
@@ -0,0 +1,108 @@
+   *
+   * @returns {Promise<Sip10AggregateBalance>}
+   */
+  async function getSip10AggregateBalance(addresses: string[], signal?: AbortSignal) {
+    const addressResults = await Promise.all(
+      addresses.map(address => getSip10AddressBalance(address, signal))
+    );
+    const totalBalance = aggregateBaseCryptoAssetBalances(addressResults.map(r => r.usd));
+    return {
+      usd: totalBalance,
+      balances: addressResults,
+    };
+  }
+
+  /**
+   * Gets all SIP10 balances for given address. Includes both cumulative and individual USD values.
+   *
+   * @returns {Promise<Sip10AddressBalance>}
+   */
+  async function getSip10AddressBalance(address: string, signal?: AbortSignal) {
+    const fungibleTokens = (await stacksApiClient.getAddressBalances(address, signal))
+      .fungible_tokens;
+    const balances = (
+      await Promise.all(
+        Object.keys(fungibleTokens).map(tokenId => {
+          return getSip10TokenBalance(
+            tokenId,
+            Number(fungibleTokens[tokenId]?.balance ?? 0),
+            signal
+          );
+        })
+      )
+    ).filter(b => !!b);
+    return {
+      address,
+      usd: aggregateBaseCryptoAssetBalances(balances.map(b => b.usd)),
+      balances,
+    };
+  }
+
+  async function getSip10TokenBalance(
+    tokenId: string,
+    amount: number,
+    signal?: AbortSignal
+  ): Promise<Sip10TokenBalance | undefined> {
+    try {
+      const tokenInfo = await sip10TokensService.getInfo(tokenId, signal);
+      const totalBalance = createMoney(amount, tokenInfo.symbol, tokenInfo.decimals);
+      const marketData = await marketDataService.getSip10MarketData(tokenInfo);
+      return {
+        info: tokenInfo,
+        sip10: createBaseCryptoAssetBalance(totalBalance),
+        usd: createBaseCryptoAssetBalance(baseCurrencyAmountInQuote(totalBalance, marketData)),
+      };
+    } catch {
+      return;
+    }
```

What's the idea behind this baked in error handling? Shouldn't this be handled outside the service? What if I want to show an error msg in the UI with the error details passed from API?

---

The idea here was to prevent the entire call from failing because of a single SIP10 failure, of which there can be many dozens under a single `getSip10Aggregate/AddressBalance`. Although I agree, this is error swallowing and probably not a great solution. I can remove.